### PR TITLE
api docs for snowflake resource

### PR DESCRIPTION
**Pull request by jamiedemaria**
### Summary & Motivation
Adds doc strings to the Snowflake resource

### CODE REVIEW

File: python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py [hunk 1]
```diff
@@ -139,7 +204,15 @@ def execute_queries(
 
         return results if fetch_results else None
 
-    def load_table_from_local_parquet(self, src, table):
+    @public
+    def load_table_from_local_parquet(self, src: str, table: str):
```
Comment (jamiedemaria) [inline python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py:208]: @gibsondan git blame says you were the last one to work on this fn, can you confirm that the doc string is accurate?

File: python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py [hunk 2]
```diff
@@ -74,8 +79,20 @@ def __init__(self, config: Mapping[str, str], log):  # pylint: disable=too-many-
         self.autocommit = self.conn_args.get("autocommit", False)
         self.log = log
 
+    @public
```
Comment (jamiedemaria) [inline python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py:82]: @smackesey I'm still getting a bit used to what should/should not get teh `@public` decorator. This was the only way I got these functions to appear in the API docs, but want to confirm that it's ok to be marking these public

File: python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py [hunk 3]
```diff
@@ -74,8 +79,20 @@ def __init__(self, config: Mapping[str, str], log):  # pylint: disable=too-many-
...
+        """Gets a connection to Snowflake as a context manager.
+
+        If using the execute_query, execute_queries, or load_table_from_local_parquet methods,
+        you do not need to create a
+        connection using this context manager.
+
+        Args:
+            raw_conn: If using the sqlalchemy connector, set raw_conn to True to create a raw
+                connection. Defaults to True.
+
```
Comment (jamiedemaria) [inline python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py:94]: TODO - add an example here, but need to find a fn on the connection that isn't executing a query

File: python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py [hunk 4]
```diff
@@ -94,7 +112,36 @@ def get_connection(self, raw_conn=True):
                 conn.commit()
             conn.close()
 
-    def execute_query(self, sql, parameters=None, fetch_results=False, use_pandas_result=False):
+    @public
+    def execute_query(
+        self,
+        sql: str,
+        parameters: Optional[Mapping[Any, Any]] = None,
```
Comment (jamiedemaria) [inline python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py:119]: TODO - see if i can make this type annotation more specific - need to dig into snowflake docs

File: python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py [hunk 5]
```diff
@@ -94,7 +123,37 @@ def get_connection(self, raw_conn=True):
                 conn.commit()
             conn.close()
 
-    def execute_query(self, sql, parameters=None, fetch_results=False, use_pandas_result=False):
+    @public
+    def execute_query(
+        self,
+        sql: str,
+        parameters: Optional[Union[Sequence[Any], Mapping[Any, Any]]] = None,
```
Comment (jamiedemaria) [inline python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py:130]: this type annotation is pulled from the snowflake type annotation https://github.com/snowflakedb/snowflake-connector-python/blob/0b24148ec0c947764c50f093f6bc1fe1638fb551/src/snowflake/connector/cursor.py#L572

**Comment by jamiedemaria** [comment]
@gibsondan  and @smackesey  bump on this when you get a minute! thanks!

File: python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py (hunk 2 above)
Comment (sryza) [reply python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py:83]: These are the rules that govern our guarantees for public APIs: https://docs.dagster.io/getting-started/releases

File: python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py (hunk 1 above)
Comment (sryza) [reply python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py:232]: This looks right to me

**Review by sryza — APPROVED** [review]
Nice!

File: python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py [hunk 6]
```diff
@@ -23,6 +24,11 @@
 
 
 class SnowflakeConnection:
+    """A connection to Snowflake that can provide a Snowflake connection and execute queries. In
```
[the remark this answers is not in the document]
Comment (jamiedemaria) [reply python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py:27]: yeah totally is

**Pull request merged**